foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/platform/database/DerbyPlatform.java

```diff
@@ -533,43 +535,73 @@ private void printQuarterJava(final Expression first, final ExpressionJavaPrinte
 
             @Override
             public void printDuo(Expression first, Expression second, ExpressionSQLPrinter printer) {
-                if (second instanceof LiteralExpression && "QUARTER".equals(((LiteralExpression)second).getValue().toUpperCase())) {
-                    printQuarterSQL(first, printer);
-                } else {
-                    super.printDuo(first, second, printer);
+                if (second instanceof LiteralExpression) {
+                    switch (((LiteralExpression) second).getValue().toUpperCase()) {
```

Comment:
make the switch complete, add `@SuppressWarnings`

Reply:
I redesigned whole EXTRACT operator code and moved those switches to only a single class - parent of all platform specific child classes. Addd @SuppressWarnings there.